cliquet/__init__.py

```diff
@@ -122,272 +101,31 @@ def load_default_settings(config):
-                alert['code'] = "soft-eol"
-                response = handler(request)
-            else:
-                response = errors.http_error(
-                    HTTPGone(),
-                    errno=errors.ERRORS.SERVICE_DEPRECATED,
-                    message=deprecation_msg)
-                alert['code'] = "hard-eol"
-            response.headers['Alert'] = utils.json.dumps(alert)
-            return response
-        return handler(request)
-    return eos_tween
-
-
-def handle_statsd(config):
-    settings = config.get_settings()
-
-    if settings['cliquet.statsd_url']:
-        client = statsd.load_from_config(config)
-
-        client.watch_execution_time(config.registry.cache, prefix='cache')
-        client.watch_execution_time(config.registry.storage, prefix='storage')
-
-        policy = config.registry.queryUtility(IAuthenticationPolicy)
-        client.watch_execution_time(policy, prefix='authentication')
-
-        def on_new_response(event):
-            request = event.request
-
-            # Count unique users.
-            user_id = request.authenticated_userid
-            if user_id:
-                client.count('users', unique=user_id)
-
-            # Count authentication verifications.
-            if hasattr(request, 'auth_type'):
-                client.count('%s.%s' % ('auth_type', request.auth_type))
-
-            # Count view calls.
-            pattern = request.matched_route.pattern
-            services = request.registry.cornice_services
-            service = services.get(pattern)
-            if service:
-                client.count('view.%s.%s' % (service.name, request.method))
-
-        config.add_subscriber(on_new_response, NewResponse)
-
-        return client
-
-
 def includeme(config):
-    # Monkey Patch Cornice Service to setup the global CORS configuration.
-    Service.cors_origins = ('*',)
-    Service.default_cors_headers = ('Backoff', 'Retry-After', 'Alert')
-
-    monkey_patch_json(config)
-
     load_default_settings(config)
     settings = config.get_settings()
 
-    # Configure cliquet logging.
-    cliquet_logs.setup_logging(config)
-
-    force_requests_url(config)
-    handle_api_redirection(config)
-    config.add_tween("cliquet.end_of_life_tween_factory")
-
-    storage = config.maybe_dotted(settings['cliquet.storage_backend'])
-    config.registry.storage = storage.load_from_config(config)
-
-    cache = config.maybe_dotted(settings['cliquet.cache_backend'])
-    config.registry.cache = cache.load_from_config(config)
-
-    id_generator = config.maybe_dotted(settings['cliquet.id_generator'])
-    config.registry.id_generator = id_generator()
-
-    set_auth(config)
-    attach_http_objects(config)
+    # Monkey Patch Cornice Service to setup the global CORS configuration.
+    # XXX: Refactor @crud decorator and inherit Service instead.
+    cors_origins = settings['cliquet.cors_origins']
+    Service.cors_origins = (cors_origins,)
```

Comment:
Well, having something imutable here is something I think we should keep. `tuple(aslist(cors_origins))` then :)